### PR TITLE
`swift build --doctor`

### PR DESCRIPTION
**Pull request by mxcl**
A command to help users troubleshoot and us to diagnose their troubles.

Currently barely anything is there, but we can add more as we go.

### CODE REVIEW

**Comment by mxcl** [comment]
@swift-ci Please test and merge

File: Sources/swift-build/--doctor.swift [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import POSIX
+
+func doctor() {
+    print("LD:", getenv("LD") ?? "nil")
+    print("which ld:", which("ld"))
+}
+
+private func which(_ cmd: String) -> String {
+    return (try? popen(["which", "ld"]))?.chomp() ?? "nil"
```
Comment (kostiakoval) [inline Sources/swift-build/--doctor.swift]: cmd is not used anywhere, maybe you wanted to run `popen(["which", cmd])`

Comment (mxcl) [inline Sources/swift-build/--doctor.swift]: oops

**Comment by mxcl** [comment]
@swift-ci please test and merge

**Pull request merged**